packages/ui/src/components/FileUpload/index.tsx

```diff
@@ -0,0 +1,189 @@
+/** *******************************************************************************************************************
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+  
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  
+      http://www.apache.org/licenses/LICENSE-2.0
+  
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.                                                                              *
+ ******************************************************************************************************************** */
+import FormField, { FormFieldProps } from '@cloudscape-design/components/form-field';
+import Button from '@cloudscape-design/components/button';
+import TokenGroup, { TokenGroupProps } from '@cloudscape-design/components/token-group';
+import StatusIndicator from '@cloudscape-design/components/status-indicator';
+import { FC, useCallback, useRef, useMemo, useState } from 'react';
+import { FileMetadata } from './types';
+import useUniqueId from '../../hooks/useUniqueId';
+import FileTokenLabel from './components/FileTokenLabel';
+import SpaceBetween from '@cloudscape-design/components/space-between';
+import getDisplaySize from './utils/getDisplaySize';
+import getDisplayLastModified from './utils/getDisplayLastModified';
+import { NonCancelableEventHandler } from '@cloudscape-design/components/internal/events';
+
+export interface FileUploadProps extends FormFieldProps {
+    /**
+     * Indicating whether to enable users to upload multiple files.
+     */
+    multi?: boolean;
+    /**
+     * One or more <a herf='https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#unique_file_type_specifiers' target='_blank'>
+     * unique file type specifiers</a> describing file types to allow
+     */
+    accept?: string;
+    /**
+     * The label text displayed on choose file button.
+     */
+    buttonText?: string;
+    /**
+     * The name of the input used in HTML forms.
+     * */
+    name?: string;
+    /**
+     * The list of choosen files.
+     */
+    files?: (File | FileMetadata)[];
```

Comment:
IDK but maybe will be good idea to create a separate type for `(File | FileMetadata)[]` - just to reduce the same type in multiple places